ui/plugin-system/src/runtime/TimeRangeProvider/TimeRangeProvider.tsx

```diff
@@ -56,37 +57,104 @@ export function useTimeRange(): TimeRange {
   return useTimeRangeContext();
 }
 
+/**
+ * Build the time range context from some initial state.
+ * The values inside the initial state can be different depending on if the query params has been enabled and/or if we
+ * gave some initial time range / refresh interval.
+ * @param ctx
+ */
+function useContextMemo(
+  ctx: Omit<TimeRange, 'refreshKey' | 'absoluteTimeRange' | 'refresh' | 'refreshIntervalInMs'>
+): TimeRange {
+  const [refreshCounter, setRefreshCounter] = useState(0);
+
+  const refresh = useCallback(() => {
+    setRefreshCounter(refreshCounter + 1);
+  }, [refreshCounter]);
+
+  return useMemo(() => {
+    const absoluteTimeRange = isRelativeTimeRange(ctx.timeRange) ? toAbsoluteTimeRange(ctx.timeRange) : ctx.timeRange;
+    return {
+      ...ctx,
+      refresh,
+      absoluteTimeRange,
+      refreshKey: `${absoluteTimeRange.start}:${absoluteTimeRange.end}:${ctx.refreshInterval}:${refreshCounter}`,
+      refreshIntervalInMs: getRefreshIntervalInMs(ctx.refreshInterval),
+    };
+  }, [ctx, refresh, refreshCounter]);
+}
+
+/**
+ * Utils function to transform a refresh interval in {@link DurationString} format into a number of ms.
+ * @param refreshInterval
+ */
+function getRefreshIntervalInMs(refreshInterval?: DurationString): number {
+  let refreshIntervalInMs = 0;
+  if (refreshInterval) {
+    const refreshIntervalDuration = parseDurationString(refreshInterval);
+    if (refreshIntervalDuration && refreshIntervalDuration.seconds) {
+      refreshIntervalInMs = refreshIntervalDuration?.seconds * 1000;
```

Comment:
```suggestion
      refreshIntervalInMs = milliseconds(refreshIntervalDuration);
```
And if condition can be removed

Reply:
Where this milliseconds is supposed to come from ??
![image](https://github.com/perses/perses/assets/10258608/73ae3e30-ee8b-421f-8d7e-8be6de3272d2)